### PR TITLE
Test merging feature/spring-boot-3 branch to main branch

### PR DESCRIPTION
**Pull request by rujche**
# Description

Draft PR, Don't review.
This PR is used to test merging **feature/spring-boot-3** branch into **main** branch.

**Preliminary conclusion:**
The external dependencies managed by Spring Boot 3 cannot be used by all SDKs and make all pipelines work normally.
The following problems have been found:
1. sdk/core, sdk/clientcore module tests depend on javax.servlet
2. Some external dependencies are already jdk11 and jdk17 compilation requirements, such as: `mockito-core: 5.11.0`; `org.eclipse.jetty.ee10:jetty-ee10-servlet:12.0.9`; `org.eclipse.jetty.ee10.websocket:jetty-ee10-websocket-jetty-server:12.0.9`
3. [slf4j-api: 2.0.13](https://github.com/qos-ch/slf4j/releases/tag/v_2.0.13) module has deprecated the class `org.slf4j.helpers.MarkerIgnoringBase` has no replacement, see https://github.com/qos-ch/logback/discussions/700
4 [Reactor-core: 3.6.6](https://github.com/reactor/reactor-core/releases/tag/v3.6.6) bug, see https://github.com/reactor/reactor-core/issues/3839

I am going to use the versions managed by Spring Boot 3 for 'sdk/spring` external dependencies, and try to make all pipelines work normally. See more from the new PR https://github.com/Azure/azure-sdk-for-java/pull/41026

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -865,6 +868,10 @@
 /eng/versioning/                                     @alzimmermsft @samvaity @g2vinay @JimSuplizio
 /eng/versioning/external_dependencies.txt            @alzimmermsft @samvaity @g2vinay @jonathangiles @rujche @netyyyy @saragluna @moarychan
 
+# Removing owners for these files to not require codeowner approval for changes to them given they are shared with project
+/eng/versioning/version_client.txt
+/eng/versioning/version_data.txt
+
```
Comment (rujche) [inline .github/CODEOWNERS:874]: These lines also exist in main branch. I don't know why the diff display here.

File: eng/versioning/version_client.txt [hunk 2]
```diff
@@ -186,7 +186,6 @@ com.azure:azure-security-keyvault-perf;1.0.0-beta.1;1.0.0-beta.1
 com.azure:azure-sdk-template;1.1.1234;1.2.2-beta.1
 com.azure:azure-sdk-template-two;1.0.0-beta.1;1.0.0-beta.1
 com.azure:azure-sdk-template-three;1.0.0-beta.1;1.0.0-beta.1
-com.azure:azure-spring-data-cosmos;3.46.0;3.47.0-beta.1
```
Comment (rujche) [inline eng/versioning/version_client.txt:189]: Moved to line 263.

File: sdk/spring/spring-reference.yml [hunk 3]
```diff
@@ -1,1519 +0,0 @@
-# Please refer to this link to see design details and how to change:
-# https://dev.azure.com/SpringOnAzure/Spring%20on%20Azure/_wiki/wikis/spring-integration-private.wiki/280/Spring-Reference-for-Intellij-Plugin
----
```
Comment (rujche) [inline sdk/spring/spring-reference.yml:3]: Hi, @saragluna , is this fil still been used? If it's still been used, I'll revert this file.

**Comment by rujche** [comment]
FYI:
After merge **main** branch to **feature/spring-boot-3-prepare-merge-to-main** branch.
The codes in **sdk/spring** folder is sync to **feature/spring-boot-3** branch.

Here is how I confirm this by Intellij IDEA:
1. Compare **sdk/spring** folder in **feature/spring-boot-3-prepare-merge-to-main** and **feature/spring-boot-3** branch:
> image

2. Compare result:
> image
> image

**Comment by rujche** [comment]
Hi, @anuchandy

### My purpose
Could you please help to fix the unit test failure in [current PR's pipeline](https://dev.azure.com/azure-sdk/public/_build/results?buildId=3898369&view=logs&j=276460ad-709f-504f-8509-6f9c6db7719d&t=95fced88-4fbf-52a9-b772-455e137c195f&l=112):

> WindowedSubscriberFluxWindowTest.shouldCancelWorkOnEmptyWindowTruncation:405 expected: <true> but was: <false>

### More details
1. This PR is used to merge **feature/spring-boot-3** branch into **main** branch.
2. The only change in **sdk/core** folder is external dependency version update.
3. Because there is too many files (1300+) changed in this PR, the [Files Changed](https://github.com/Azure/azure-sdk-for-java/pull/40700/files) tab in current page is hard to use. You can use **Intellij IDEA** to view the changes. Here are the screenshots about how to do this:
> image
> image

### Another topic
This is another topic, if you don't know the answer, please skip this part, it's not necessary to spend much time to investigate this.

To let you easy reproduce the issue, I created another PR: #40756, which just change the pom files in sdk/core folder like this PR did. But [the pipeline](https://dev.azure.com/azure-sdk/public/_build/results?buildId=3893489&view=logs&j=41c6a66b-cae9-587e-a8fd-e2e15f25240c&t=51a7b4bf-9d43-5dee-c62f-b2d1156299de&l=25) failed with another issue. It failed in Build for **non-From Source run**, so the **Run Tests** is skipped. Here are the screenshots:
> image

I confirmed that the maven command is exactly the same:
> image
> image

I don't know why one is failed by warnings and another one is not:
> image

As I said before, please help to fix the pipeline failure first, ignore the **Another topic** part if you don't know the reason.

**Comment by Netyyyy** [comment]
@rujche remove `EnableBatchRelease` to run ci pipeline because it's template already removed https://github.com/Azure/azure-sdk-for-java/pull/40742/commits/f4b9fc34053c8e97044968fc6cfd01c7611a067a

**Comment by anuchandy** [comment]
> Hi, @anuchandy
> 
> ### My purpose
> Could you please help to fix the unit test failure in [current PR's pipeline](https://dev.azure.com/azure-sdk/public/_build/results?buildId=3898369&view=logs&j=276460ad-709f-504f-8509-6f9c6db7719d&t=95fced88-4fbf-52a9-b772-455e137c195f&l=112):
> 
> > WindowedSubscriberFluxWindowTest.shouldCancelWorkOnEmptyWindowTruncation:405 expected:  but was:
> 
> ### More details
> 1. This PR is used to merge **feature/spring-boot-3** branch into **main** branch.
> 2. The only change in **sdk/core** folder is external dependency version update.
> 3. Because there is too many files (1300+) changed in this PR, the [Files Changed](https://github.com/Azure/azure-sdk-for-java/pull/40700/files) tab in current page is hard to use. You can use **Intellij IDEA** to view the changes. Here are the screenshots about how to do this:
> 
> > image
> > image
> 
> ### Another topic
> This is another topic, if you don't know the answer, please skip this part, it's not necessary to spend much time to investigate this.
> 
> To let you easy reproduce the issue, I created another PR: #40756, which just change the pom files in sdk/core folder like this PR did. But [the pipeline](https://dev.azure.com/azure-sdk/public/_build/results?buildId=3893489&view=logs&j=41c6a66b-cae9-587e-a8fd-e2e15f25240c&t=51a7b4bf-9d43-5dee-c62f-b2d1156299de&l=25) failed with another issue. It failed in Build for **non-From Source run**, so the **Run Tests** is skipped. Here are the screenshots:
> 
> > image
> 
> I confirmed that the maven command is exactly the same:
> 
> > image
> > image
> 
> I don't know why one is failed by warnings and another one is not:
> 
> > image
> 
> As I said before, please help to fix the pipeline failure first, ignore the **Another topic** part if you don't know the reason.

Hello @rujche, @moarychan, I looked into this and seems a regression in reactor-core 3.6.6, reported it here https://github.com/reactor/reactor-core/issues/3839

**Comment by moarychan** [comment]
Hi @alzimmermsft , there's some build failures for deprecation usage, I tried to copy the class `MarkerIgnoringBase`, and now there's new failure for test `com.azure.core.http.policy.HttpLoggingPolicyTests#loggingIncludesRetryCountSync` . Could you help provide a replacement for class `MarkerIgnoringBase` and make the test passed? please find the [pipeline test failures](https://dev.azure.com/azure-sdk/public/_build/results?buildId=3930946&view=logs&j=41c6a66b-cae9-587e-a8fd-e2e15f25240c&t=f9858f1c-ee5b-5d74-57ed-56b704cc3a33&l=357) in another PR https://github.com/Azure/azure-sdk-for-java/pull/40756. Thanks.

**Comment by moarychan** [comment]
I have made a summary in the PR description, closing it now.

**Pull request closed without merging**